specification/columns/commitmentdiscountunit.md

```diff
@@ -0,0 +1,36 @@
+# Commitment Discount Unit
+
+Commitment Discount Unit is the string value representing the provider-specified measurement unit that corresponds to [Commitment Discount Quantity](#commitmentdiscountconsumedquantity) and [Commitment Discount Purchased Quantity](#commitmentdiscountpurchasedquantity) of a [*commitment discount*](#glossary:commitment-discount). The CommitmentDiscountUnit column is only applicable to *commitment discounts* and not [*negotiated discounts*](#glossary:negotiated-discount).
+
+The CommitmentDiscountUnit column adheres to the following requirements:
+
+* CommitmentDiscountUnit MUST be present in the billing data when the provider supports [*commitment discounts*](#glossary:commitment-discount).
+* CommitmentDiscountUnit MUST be of type String, and the units of measure used in CommitmentDiscountUnit SHOULD adhere to the values and format requirements specified in the [UnitFormat](#unitformat) attribute.
```

Comment:
Shouldn't we mandate that this use the same value as ConsumedUnit for usage commitments? Should we mandate that it be the BillingCurrency for spend commitments? If not, what would it be for spend commitments?
```suggestion
* CommitmentDiscountUnit MUST be of type String, and the units of measure used in CommitmentDiscountUnit SHOULD adhere to the values and format requirements specified in the [UnitFormat](#unitformat) attribute.
* When [ChargeCategory](#chargecategory) is "Usage" and [CommitmentDiscountCategory](#commitmentdiscountcategory) is "Usage", CommitmentDiscountUnit MUST use the same value as [ConsumedUnit](#consumedunit).
* When [ChargeCategory](#chargecategory) is "Usage" and [CommitmentDiscountCategory](#commitmentdiscountcategory) is "Spend", CommitmentDiscountUnit MUST use the same value as [BillingCurrency](#billingcurrency).
```

Reply:
With size flex, "Normalized Hours" are neither the `ConsumedUnit` or the `BillingCurrency`

Reply:
Does the Unit Format attribute have an affordance for "Normalized"? That might be worth calling out there. Either way, if we need to account for normalization, we can easily add that to the requirements. Then we can say it SHOULD be "Normalized Foo" when normalization is applied and matching ConsumedUnit when it's not normalized.

Reply:
Follow-up on suggestion by @flanakin:

#### Commitment Discount Category: Spend

- This allows us to limit Spend to BillingCurrency and exclude non-currency consumable units (NCCUs), as @rileyjenk calls them? Providers can still support NCCUs and use these columns if all other commitment discount requirements are met, but they need to convert NCCUs into BillingCurrencies.
- If we decide to impose this, CommitmentDiscountConsumedQuantity would align with EffectiveCost in the case of Usage, and CommitmentDiscountPurchasedQuantity would align with BilledCost in case of Purchase. I’m not suggesting that explicitly stating this adds any value; I’m simply pointing this out.

#### Commitment Discount Category: Usage

- Normalized Hours are essentially Hours, and by calling them "Normalized," we imply that the value in CommitmentDiscountConsumedQuantity is normalized? I would prefer to use the same value for both CommitmentDiscountUnit and ConsumedUnit and specify that CommitmentDiscountConsumedQuantity includes the size normalization factor (size normalization factor should be added to the glossary).
- Using the same value for both CommitmentDiscountUnit and ConsumedUnit would help practitioners easily detect charges with normalization factors by comparing CommitmentDiscountConsumedQuantity and ConsumedQuantity, and even calculate the factor by dividing these quantities.

- **NOTE:** Keep in mind that we have specified that for Unused Usage-related records, CommitmentDiscountUnit MUST NOT be null, but ConsumedUnit MUST be null. If we choose this approach, we need to **either rephrase the requirement or revert to having ConsumedQuantity and ConsumedUnit not be null in the case of Unused Usage**.